Content.Client/Access/UI/AgentIDCardWindow.xaml.cs

```diff
@@ -1,24 +1,96 @@
-using Robust.Client.UserInterface.CustomControls;
+using Content.Client.Stylesheets;
+using Content.Shared.StatusIcon;
 using Robust.Client.AutoGenerated;
+using Robust.Client.GameObjects;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
 using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+using System.Linq;
+using System.Numerics;
 
 namespace Content.Client.Access.UI
 {
     [GenerateTypedNameReferences]
     public sealed partial class AgentIDCardWindow : DefaultWindow
     {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IEntitySystemManager _entitySystem = default!;
+        private readonly SpriteSystem _spriteSystem = default!;
+
+        private const string JobIconPrarentId = "JobIcon";
+        private const int JobIconColumnCount = 10;
+
         public event Action<string>? OnNameChanged;
         public event Action<string>? OnJobChanged;
+        public readonly List<Button> JobIconButtons = new();
+        public readonly Dictionary<int, string> StateByIndex = new();
 
         public AgentIDCardWindow()
         {
             RobustXamlLoader.Load(this);
+            IoCManager.InjectDependencies(this);
+            _spriteSystem = _entitySystem.GetEntitySystem<SpriteSystem>();
 
             NameLineEdit.OnTextEntered += e => OnNameChanged?.Invoke(e.Text);
             NameLineEdit.OnFocusExit += e => OnNameChanged?.Invoke(e.Text);
 
             JobLineEdit.OnTextEntered += e => OnJobChanged?.Invoke(e.Text);
             JobLineEdit.OnFocusExit += e => OnJobChanged?.Invoke(e.Text);
+
+            var jobIconGroup = new ButtonGroup();
+            var allJobIcons = _prototypeManager.EnumeratePrototypes<StatusIconPrototype>();
+            var i = 0;
+            foreach (var jobIcon in allJobIcons)
+            {
+                if (!(jobIcon.Parents?.Contains(JobIconPrarentId) ?? false))
+                {
+                    continue;
+                }
+
+                var specifier = jobIcon.Icon as SpriteSpecifier.Rsi;
+                if (specifier == null)
+                {
+                    continue;
+                }
```

Comment:
just use `_spriteSystem.Frame0(jobIcon.Icon),`. This adds an arbitrarily hidden restriction that icon prototypes need to all use RSIs. If, for whatever reason, the restriction is actually needed, then the icon data-field needs to be changed so a `SpriteSpecifier.Rsi` instead of a `SpriteSpecifier` field